src/util/taproot.rs

```diff
@@ -760,16 +764,16 @@ impl ControlBlock {
 /// The leaf version for tapleafs
 #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-pub struct LeafVersion(u8);
+pub enum LeafVersion {
```

Comment:
Just realized this changes serde (de)serialization. I think we should still (de)serialize as a single number at least for non-human-friendly encodings.

Reply:
Good catch! I agree.